app/src/main/java/ch/hikemate/app/ui/components/WithDetailedHike.kt

```diff
@@ -37,58 +47,73 @@ fun WithDetailedHike(
     hike: Hike,
     hikesViewModel: HikesViewModel,
     withDetailedHike: @Composable (DetailedHike) -> Unit,
-    whenError: @Composable () -> Unit
+    whenError: @Composable () -> Unit,
+    navigationActions: NavigationActions,
+    onBackAction: () -> Unit
 ) {
   val loadingErrorMessageId by hikesViewModel.loadingErrorMessageId.collectAsState()
 
-  when {
-    // All the details of the hike have been computed, pass them on to the content callback
-    hike.isFullyLoaded() -> {
-      val detailedHike =
-          try {
-            hike.withDetailsOrThrow()
-          } catch (e: Exception) {
-            Log.d(WithDetailedHike.LOG_TAG, "Error while getting detailed hike: $e")
-            null
-          }
+  var detailedHike: DetailedHike? by remember { mutableStateOf(null) }
```

Comment:
Question: Why do you need this variable? It seems it could work without it

Reply:
Yep, it works without it